packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts

```diff
@@ -1007,50 +1021,133 @@ describe('ModelInstanceDocumentHandler', () => {
     expect(state2).toMatchSnapshot()
   })
 
-  it('Rejects commit if field is immutable', async () => {
-    const genesisCommit = (await ModelInstanceDocument._makeGenesis(context.signer, CONTENT0, {
-      controller: DID_ID,
-      model: FAKE_MODEL_IMMUTABLE_ID,
-    })) as SignedCommitContainer
-    await ipfs.dag.put(genesisCommit, FAKE_CID_1)
-
-    const payload = dagCBOR.decode(genesisCommit.linkedBlock)
-    await ipfs.dag.put(payload, genesisCommit.jws.link)
-
-    // apply genesis
-    const genesisCommitData = {
-      cid: FAKE_CID_1,
-      type: CommitType.GENESIS,
-      commit: payload,
-      envelope: genesisCommit.jws,
-    }
-
-    const state = await handler.applyCommit(genesisCommitData, context)
+  describe('Immutable checks', () => {
```

Comment:
can we also add a test that tries to change the type of a field where the schema allows multiple types, but where the value will look similar?  E.g. a field that allows a string or a number, where you try to change it from `1` (number) to `"1"` (string)?